### DISCUSSION TITLE
Reuse receive buffer with FramedRead

### DISCUSSION THREAD

**Discussion by erenon** [tokio-rs/tokio#6579] [Q&A] [answered]
Hi,
I can create a FramedRead with LengthDelimitedCodec, and read messages using `framed_read.next().await??`. The message type given that way is a `BytesMut`. This means, for each message, a new buffer is allocated. Instead, I'd like to get a reference to the underlying buffer, process that, and re-use the allocated memory for the next read. Is that possible with FramedRead? thanks.

**Comment by Darksonn** [comment, marked as answer]
Actually, it is not necessarily the case that a new buffer is allocated for each message. If you make sure to drop the `BytesMut` before you call `next` again, then when it calls `BytesMut::reserve`, it will notice that the refcount of the `BytesMut` is 1 which allows it to reuse the existing allocation.

**Reply by erenon** [reply]
Nice, thanks!
